### DISCUSSION TITLE
When omiting `message` from `setError`, is it possible to use an already registered message?

### DISCUSSION THREAD

**Discussion by luisdralves** [react-hook-form/react-hook-form#7647] [Q&A] [answered]
For an input such as this:
```tsx
<input
  name={'name'}
  {...register('name', {
    required: translate('common:form.rules.required')
  })}
/>
```

I would like to be able to call `setError` without having to pass the message again, like such:
```ts
setError('name', { type: 'required' })
```

But still having the message in the `errors` object:
```ts
{
  errors: {
    name: {
      message: '...' // the resolved string from translate('common:form.rules.required')
      type: 'required'
    }
  }
}
```

Instead, following these steps, only the type is passing through. Is there any way to get the registered error messages? I was able to get them though the `control` properties, but the documentation says not to do that.

**Comment by Moshyfawn** [comment, marked as answer]
Wouldn't [the `trigger` form method](https://react-hook-form.com/api/useform/trigger) do what you want?
Calling `trigger` with the field's name would trigger it's validation rules check and set the error messages you've already set during registering the input
```js
<input
  name={'name'}
  {...register('name', {
    required: translate('common:form.rules.required')
  })}
/>
``` 
```js
trigger('name')
```

**Reply by luisdralves** [reply]
Not in this case, no. It is not the kind of validation that can be done client-side. I need to trigger the errors based on the response of a POST/PATCH request. For example, if some fields must be unique, the response code is `409: Conflict` with an array of the fields that must be changed.

After giving it some thought with the rest of the team we came to the conclusion that there isn't much overlap between client-side and server-side error messages, so we will define the server-side messages elsewhere. There is no need to do what I asked about in the first place.

**Reply by Moshyfawn** [reply]
You can still define your [custom validate callback](https://react-hook-form.com/api/useform/register) with some `async` logic _(requests, computations etc.)_ and [`trigger`](https://react-hook-form.com/api/useform/trigger) it whenever.
```js
<input {...register("name", { validate: validateName })} />
<button onClick={() => trigger("name")}>Trigger Validation</button>
```
See [this CSB](https://codesandbox.io/s/optimistic-pare-gr3fc?file=/src/App.tsx) for an example with an `onSubmit` and `trigger` validation

**Reply by luisdralves** [reply]
I see, this totally fits my use-case. Thank you very much.

**Reply by Moshyfawn** [reply]
Awesome, np!
